agents/appmotion_motor/sat1_motors_driver.py

```diff
@@ -0,0 +1,825 @@
+# https://www.applied-motion.com/sites/default/files/hardware-manuals/Host-Command-Reference_920-0002P.PDF
+#
+####################################################################################################
+
+
+import sys
+from socs.agent.moxaSerial import Serial_TCPServer
+from time import sleep
+import numpy as np
+import traceback
+
+
+# Time to wait for the user to power on the controllers and to see the power-on signature from the serial port
+DEFAULT_WAIT_START_TIME = 15.0 #seconds
+
+# Motor Names
+MOTOR1 = 1
+MOTOR2 = 2
+ALL = 3
+
+# Conversions for stages
+AXIS_THREADS_PER_INCH_STAGE = 10.0 #Conversion for the FTS Linear Stage - Check for factor of two later
+AXIS_THREADS_PER_INCH_XYZ = 10.0 #Measured on stepper, Check for factor of two later
+#AXIS_THREADS_PER_INCH = 10.0 #Measured on stepper
+
+
+
+class MotControl(object):
+        """Object for controlling up to 2 motors - all functions will work for linear stages, some will work if motors are attached
+        to non-linear stages. 
+        """
+
+        def __init__(self, motor1_Ip=None, motor1_Port=None, motor1_isLin = True, motor2_Ip=None, motor2_Port=None, motor2_isLin = True, mRes=False):
+                """Initialize a MotControl object.
+
+                Parameters:
+                UPDATE********************
+                motor1_Ip (str) -- the IP address associated with the linear stage motor
+                motor1_Port (int) -- the port address associated with the linear stage motor
+                motor2_Ip (str) -- the IP address associated with the output polarizer motor
+                motor2_Port (int) -- the port address associated with the output polarizer motor
+                mRes (bool) -- True if manual resolution, False if default (res=8) ???
+                """
+
+                # Set up the connection to the first motor
+                # Initialized so that the startup position is set to zero
+                if not (motor1_Ip and motor1_Port):
+                        print("Invalid Motor 1 information.  No Motor 1 control.")
+                        self.motor1 = None
+                else:
+                        print('establishing serial server with motor1!')
+                        self.motor1 = Serial_TCPServer((motor1_Ip, motor1_Port))
+                        if mRes:
+                                self.motor1.propDict = {
+                                        'name': 'motor1',
+                                        'motor': MOTOR1,
+                                        'res': 'manual',
+                                        'pos': 0, #Position in counts (should always be an integer)
+                                        'realPos': 0.0, #Position in inches
+                                        'sPRev': 8000.0, #Steps per revolution (thread)
+                                        'isLin': motor1_isLin
+                                }
```

Comment:
This construction is a bit strange. You're jamming a dictionary into an undefined attribute of the `Serial_TCPServer` object for the motors. The structure that it looks like you want is a `Motor` object that has the `Serial_TCPServer` object for communication as well as all of these entries in the `propDict` as class attributes.

That would sort of clean up references to these quasi-attributes, making lines that that are currently `motor.propDict['sPRev']` look more like `motor.sprev`. This would also reduce a lot of the repeated code when setting up motor2.

Reply:
I got rid of `propDict` and changed it so that `motor1` and `motor2` are their own objects with attributes

Reply:
That's not quite what I meant. Making a new Motor object would look more like adding a new class:
```
class Motor:
    def __init__(self, ip, port, is_lin=True, name=None, index=None, m_res=False):
        self.ip = ip,
        self.port = port,
        self.is_lin = is_lin,
        self.name = name
        self.motor = index
        self.ser = Serial_TCPServer((ip, port))

        self.pos = 0 # Position in counts (should always be an integer)
        self.real_pos = 0.0 # Position in inches

        if m_res == 'manual':
            self.res = 'manual'
            self.s_p_rev = 8000.0
        else:
            self.res = '8'
            self.s_p_rev = 20000.0
```
And then instantiating it like:
```
print('establishing serial server with motor1!')
self.motor1 = Motor(motor1_ip, motor1_port, motor1_is_lin, name='motor1', index=MOTOR1, m_res=m_res)

print('establishing serial server with motor2!')
self.motor2 = Motor(motor2_ip, motor2_port, motor2_is_lin, name='motor2', index=MOTOR2, m_res=m_res)
```

`propDict` before was being added as a new (and undefined) attribute to the `Serial_TCPServer` object, now you're adding 7 undefined attributes (`res`, `s_p_rev`, `name`, `motor`, `pos`, `real_pos`, `is_lin`) to the `Serial_TCPServer` object.